File: Data Structures & Algorithms/string-encode-and-decode/submission-0.java
class Solution {


    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        StringBuilder enStr = new StringBuilder();
        for(String str: strs) enStr.append(str.length()).append('#').append(str);
        return enStr.toString();
    }
   // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        List<String> decoded = new ArrayList<> ();
    
        for(int i=0;i<s.length();){
            int delimiter = s.indexOf('#',i);
            int strLength = Integer.parseInt(s.substring(i,delimiter));
            String word = s.substring(delimiter+1,delimiter+strLength+1);
            decoded.add(word);
            i=delimiter+strLength+1;
        }
        return decoded;
    }
}

File: Data Structures & Algorithms/longest-common-prefix/submission-0.py
class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        curr = strs[0]
        for s in strs[1:]:
            while not s.startswith(curr):
                curr = curr[:-1]
                if not curr:
                    return ""
        return curr

        